documentation/general/torn-sdk.md

```diff
@@ -0,0 +1,142 @@
+# Torn .NET SDK
+
+## Terminology
+
+- msbuild: refers the .NET Framework based msbuild included with Visual Studio.
+- dotnet msbuild: refers to the .NET Core based msbuild included with the .NET SDK.
+- analyzers: this document will use analyzers to refer to both analyzers and generators.
+
+## Summary
+
+Visual Studio and .NET SDK are separate products but they are intertwined in command line and design time build scenarios as different components are loaded from each product.
+
+| Scenario | Loads Roslyn | Loads Analyzers / Generators |
+| --- | --- | --- |
+| msbuild | From Visual Studio | From .NET SDK |
+| dotnet msbuild | From .NET SDK | From .NET SDK |
+| Visual Studio Design Time | From Visual Studio | From Both |
+
+Generally this mixing of components is fine because Visual Studio will install a .NET SDK that is functionally paired with it. That is the compiler and analyzers are the same hence mixing poses no real issue. For example 17.10 installs .NET SDK 8.0.3xx, 17.9 installs .NET SDK 8.0.2xx, etc ... However when the .NET SDK is not paired with the Visual Studio version,then compatibility issues can, and will, arise. Particularly when the .NET SDK is _newer_ than Visual Studio customers will end up with the following style of error:
+
+> CSC : warning CS9057: The analyzer assembly '..\dotnet\sdk\8.0.200\Sdks\Microsoft.NET.Sdk.Razor\source-
+generators\Microsoft.CodeAnalysis.Razor.Compiler.SourceGenerators.dll' references version '4.9.0.0' of the compiler, which is newer
+than the currently running version '4.8.0.0'.
+
+To address this issue we are going to separate Visual Studio and the .NET SDK in build scenarios. That will change our matrix to the following:
```

Comment:
You should include data to back up that this is a common problem. Here's the data I have from back in December:
![image](https://github.com/jaredpar/sdk/assets/12663534/ff83f0c9-15bb-41ec-aefc-59a1591c4f78)

Reply:
Does this telemetry include CI runs?